ql/src/test/queries/clientpositive/cte_mat_11.q

```diff
@@ -0,0 +1,15 @@
+--! qt:dataset:src
+set hive.optimize.cte.materialize.threshold=2;
+set hive.optimize.cte.materialize.full.aggregate.only=false;
+
+EXPLAIN WITH materialized_cte1 AS (
+  SELECT * FROM src
+),
+materialized_cte2 AS (
+  SELECT *
+  FROM materialized_cte1 a
+  JOIN materialized_cte1 b ON (a.key = b.key)
+)
+SELECT *
+FROM materialized_cte2 a
+JOIN materialized_cte2 b ON (a.key = b.key);
```

Comment:
I haven't dig in deeply what could be the cause of the failure. Maybe it can be a bug or it is something which Hive doesn't support and there are opportunities to make some improvement at this area or it is something which does not makes sense by SQL semantics. So it requires some investigation first. If it is a bug please file a new jira and include the repro steps.

About propagating stats: I think it is a good idea. I see some plan changes in existing tests so it seems that those are proving that the stats propagation happens. If you think we need some additional tests please feel free to add it. You can easily check whether CBO is passing or not by adding an `EXPLAIN CBO <query>` command, it should print the CBO plan.

Reply:
@kasakrisz
You find very well! I didn't notice the possibility where my test case could be illegal. It will be probably invalid as far as I tested it with various parameters or other DBMS systems. I created the following ticket and am investigating more.
https://issues.apache.org/jira/browse/HIVE-28088

Additionally, I changed `cte_mat_11.q` so that it contains only valid SQLs. I am guessing HIVE-28088 is disjunctive with HIVE-28080, and we can review this PR. Please feel free to put any other concerns.
https://github.com/apache/hive/pull/5089/commits/961f27fce8c6be78bac8d44d70c25176a57a4fe1

Thanks a lot.